src/identity/components/Socials.tsx

```diff
@@ -0,0 +1,113 @@
+import type { Address, Chain } from 'viem';
+import { githubSvg } from '../../internal/svg/githubSvg';
+import { twitterSvg } from '../../internal/svg/twitterSvg';
+import { warpcastSvg } from '../../internal/svg/warpcastSvg';
+import { websiteSvg } from '../../internal/svg/websiteSvg';
+import { border, cn, pressable } from '../../styles/theme';
+import { useName } from '../hooks/useName';
+import { useSocials } from '../hooks/useSocials';
+import { useIdentityContext } from './IdentityProvider';
+
+type SocialsReact = {
+  address?: Address | null;
+  ensName?: string;
+  chain?: Chain;
+  className?: string;
+};
+
+export function Socials({ address, chain, className }: SocialsReact) {
+  const { address: contextAddress, chain: contextChain } = useIdentityContext();
+
+  const accountAddress = address ?? contextAddress;
+  const accountChain = chain ?? contextChain;
+
+  if (!accountAddress) {
+    console.error(
+      'Socials: an Ethereum address must be provided to the Socials component.',
+    );
+    return null;
+  }
+
+  const { data: name, isLoading: isLoadingName } = useName({
+    address: accountAddress,
+    chain: accountChain,
+  });
+
+  const { data: socials, isLoading: isLoadingSocials } = useSocials(
+    {
+      ensName: name ?? '',
+      chain: accountChain,
+    },
+    { enabled: !!name },
+  );
+
+  if (isLoadingName || isLoadingSocials) {
+    return <span className={className} />;
```

Comment:
should this have some sort of loading state?

Reply:
I don't think so. Because there is a possibility that the user will not have any socials associated with the address, and in that case we will not render the Socials component. The current setup is cleaner